src/symbol.rs

```diff
@@ -70,6 +70,20 @@ macro_rules! gen_symbol_for {
             value: $non_zero,
         }
 
+        impl $name {
+            /// Creates an invalid symbol that can be used as a placeholder.
+            ///
+            /// # Safety
+            ///
+            /// Trying to resolve returned symbol is always undefined behavior.
+            #[inline]
+            pub const unsafe fn new_uninit() -> Self {
+                unsafe {
+                    Self { value: <$non_zero>::new_unchecked(0) }
```

Comment:
This is undefined behavior: https://doc.rust-lang.org/core/num/struct.NonZeroIsize.html#method.new_unchecked

Reply:
It's UB because the code using NonZero's expects them to be non-zero.

I noted under Safety that using the created value is undefined behavior and the added function is unsafe.

If the symbol is never used (Safety requirements are satisfied), construction of invalid value is sound. Similar to how creating dangling pointers is safe, this should be as well - as long as it's not "dereferenced". https://doc.rust-lang.org/std/ptr/struct.NonNull.html#method.dangling

Maybe using a transmute with 0-inited array of $non_zero size would be better as this does violate new_unchecked Safety, although it just wraps the original numbers.

Also, new_invalid might be a more appropriate name.

Reply:
Can you not simply create your own symbol type as allowed by the library already and have your use case out of the box?